.github/workflows/typespec-requirement.yaml

```diff
@@ -1,22 +1,33 @@
 name: TypeSpec Requirement
 
-on: pull_request
+on:
+  pull_request_target:
+    branches:
+      - 'main'
 
 jobs:
-  TypeSpec-Requirement:
+  typespec-requirement:
     name: TypeSpec Requirement
     strategy:
       fail-fast: false
       matrix:
         spec-type: [data-plane, resource-manager]
 
+    permissions:
+      contents: read
+
     runs-on: ubuntu-latest
 
+    outputs:
+      spec-lifecycle-data-plane: ${{ steps.tsr-ps1.outputs.spec-lifecycle-data-plane }}
```

Comment:
I assume this is a workaround given the outputs would normally stomp on each other?

Reply:
Yes, you need to manually disambiguate the outputs (and artifacts) from matrix jobs to avoid overwriting:

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/passing-information-between-jobs?apiVersion=2022-11-28#using-job-outputs-in-a-matrix-job